code/util/BaseInteractable.cs

```diff
@@ -1,11 +1,45 @@
-﻿namespace Sandbox.util;
+﻿using Cinema;
+using System.Linq;
+
+namespace Sandbox.util;
+public class Slot
+{
+    public Entity Entity { get; set; }
+    public string Attachment { get; set; }
+    public int Index { get; set; }
+
+    public int MaxDistanceTarget = 8; // This is the max distance to target the specific target
+    public Slot(int index, string attachment)
+    {
+        Index = index;
+        Attachment = attachment;
+    }
+
+    public void SetItem(Entity entity)
+    {
+        Entity = entity;
+    }
+    public void Clear()
+    {
+        Entity.Delete();
+    }
+    public bool IsEmpty()
+    {
+        return !Entity.IsValid();
+    }
+
+    public bool HasItem()
```

Comment:
Not clear name; HasItem == Entity is valid?

Maybe we rename `HasItem` => `HasEntity` and same with `SetItem`?

Reply:
Could also name it `Contents`

Reply:
Yeah, the naming is a little messy. I changed both functions to Set/Has Contents for now. I'm thinking in the next iteration, there will be some sort of class that works similar to an inventory system for interactables that contain other entities